service/src/main/java/com/commercetools/pspadapter/payone/IntegrationService.java

```diff
@@ -154,18 +168,35 @@ public int port() {
         return Integer.parseInt(systemProperty);
     }
 
-    private ImmutableMap<String, Object> createHealthResponse(@Nonnull final ServiceConfig serviceConfig) {
+    ImmutableMap<String, Object> createHealthResponse(@Nonnull final String version, @Nonnull final String title,
+                                                      List<TenantFactory> tenants) {
         final ImmutableMap<String, String> applicationInfo = ImmutableMap.of(
-                "version", serviceConfig.getApplicationVersion(),
-                "title", serviceConfig.getApplicationName());
-
-        final List<String> tenants = tenantFactories.stream()
-                .map(TenantFactory::getTenantName)
-                .collect(toList());
-
+                "version", version,
+                "title", title);
+        Map<String, CompletionStage<Integer>> tenantMap = checkTenantStatuses(tenants);
+        //resolve all completable stages
+        listOfFuturesToFutureOfList(new ArrayList<>(tenantMap.values()));
+        //unpack completable features
+        Map<String, Integer> statusMap = tenantMap.keySet().stream()
+                .collect(toMap(v -> v, v -> tenantMap.get(v).toCompletableFuture().join()));
         return ImmutableMap.of(
-                "status", HttpStatus.OK_200,
-                "tenants", tenants,
-                "applicationInfo", applicationInfo);
+                STATUS_KEY, !statusMap.containsValue(ERROR_STATUS) ? SUCCESS_STATUS : ERROR_STATUS,
+                TENANTS_KEY, statusMap,
+                APPLICATIONINFO_KEY, applicationInfo);
```

Comment:
`APPLICATION_INFO_KEY`?

Reply:
Not sure if I completely understand here, so this method returns only 1 entry? always?

Reply:
2e46afcf3

Reply:
but 2nd comment